src/components/PostDetails.tsx

```diff
@@ -1,105 +1,60 @@
-import React, { useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addComment,
+  fetchComments,
+  removeComment,
+} from '../features/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const post
+    = useAppSelector(state => state.selectedPost.selectedPost);
+  const dispatch = useAppDispatch();
+  const { items, loaded, hasError } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
     setVisible(false);
-    setError(false);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post?.id !== undefined) {
+      dispatch(fetchComments(post.id));
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
+  }
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  useEffect(loadComments, [post?.id]);
 
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
+  const handleAddComment = async ({ name, email, body }: CommentData) => {
+    if (post?.id !== undefined) {
```

Comment:
```suggestion
    if (post.id) {
```

Reply:
I can't use dectructuring here since selectedPost can be null